src/main/java/org/aion/wallet/ui/components/HistoryController.java

```diff
@@ -127,6 +151,28 @@ private void setEventHandlers() {
         copyItem.setOnAction(new ContextMenuTableCopyEventHandler(txTable));
         menu.getItems().add(copyItem);
         txTable.setContextMenu(menu);
+
+
+        searchField.textProperty().addListener((observable, oldValue, newValue) -> {
+            FilteredList<TxRow> filteredData = new FilteredList(FXCollections.observableList(completeTransactionList), s -> true);
```

Comment:
this row can be  `final FilteredList<TxRow> filteredData = new FilteredList<> (FXCollections.observableList(completeTransactionList));`
since you add the filer predicate in the conditional block below anyway